client/v2/client.go

```diff
@@ -134,41 +134,47 @@ func (c *Client) Send(data []interface{}) error {
 		// compress payload
 		w, err := zlib.NewWriterLevel(c.wb, c.opts.compressLvl)
 		if err != nil {
-			return err
+			return len(data), err
 		}
 
-		if err := c.serialize(w, data); err != nil {
-			return err
+		if dropped, err = c.serialize(w, data); err != nil {
+			return len(data), err
 		}
 
 		if err := w.Close(); err != nil {
-			return err
+			return len(data), err
 		}
 
 		// write compress header
 		payloadSz := c.wb.Len() - offPayload
 		binary.BigEndian.PutUint32(c.wb.Bytes()[offSz:], uint32(payloadSz))
 	} else {
-		if err := c.serialize(c.wb, data); err != nil {
-			return err
+		var err error
+		if dropped, err = c.serialize(c.wb, data); err != nil {
+			return len(data), err
 		}
 	}
 
+	if dropped == len(data) {
+		// all events have been dropped, we're done.
+		return dropped, nil
```

Comment:
There was no error. In case non-serializable events are dropped it is 'silent'. But we must no send empty batches to Logstash, as this could break the connection. We just report success here.